### PR TITLE
nix, nix-build, nix-env, nix-collect-garbage, nix-shell: add German translation

### PR DESCRIPTION
**Pull request by rockofox**
- [x] The page(s) are in the correct platform directories: `common`, `linux`, `osx`, `windows`, `sunos`, `android`, etc.
- [x] The page(s) have at most 8 examples.
- [x] The page description(s) have links to documentation or a homepage.
- [x] The page(s) follow the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).
- **Version of the command being documented (if known):**

### CODE REVIEW

File: pages.de/common/nix-build.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# nix-build
+
+> Erstellen eines Nix-Ausdruck.
+> Weitere Informationen: <https://nixos.org/releases/nix/latest/manual#sec-nix-build>.
+
+- Erstellen eines Nix-Ausdrucks:
+
+`nix-build --attr {{ausdruck_name}}`
```
Comment (EmilySeville7cfg) [inline pages.de/common/nix-build.md:8]: The following fix could be applied after the [original](https://github.com/tldr-pages/tldr/blob/main/pages/common/nix-build.md) page is updated:

> Concrete `ausdruck_name` value should be provided.

It's better to use certain values.

File: pages.de/common/nix-build.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+- Erstellen eines Nix-Ausdruck mit Sandbox (auf nicht-NixOS):
+
+`nix-build --attr {{ausdruck_name}} --option sandbox true`
```
Comment (EmilySeville7cfg) [inline pages.de/common/nix-build.md:12]: The following fix could be applied after the [original](https://github.com/tldr-pages/tldr/blob/main/pages/common/nix-build.md) page is updated:

> Concrete `ausdruck_name` value should be provided.

It's better to use certain values.

File: pages.de/common/nix-collect-garbage.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+`sudo nix-collect-garbage --delete-old`
+
+- Simulieren der Löschung alter Speicherpfade:
+
+`sudo nix-collect-garbage --delete-old --dry-run`
+
+- Löschen aller Speicherpfade, die älter als 30 Tage sind:
+
+`sudo nix-collect-garbage --delete-older-than {{30d}}`
```
Comment (EmilySeville7cfg) [inline pages.de/common/nix-collect-garbage.md:17]: The following fix could be applied after the [original](https://github.com/tldr-pages/tldr/blob/main/pages/common/nix-collect-garbage.md) page is updated:

```suggestion
`sudo nix-collect-garbage --delete-older-than 30d`
```

- Placeholders can be omitted for certain values.
- Our style guide doesn't specify whether this fix should be applied or not.

File: pages.de/common/nix-env.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+`nix-env -q {{search_term}}`
+
+- Verfügbare Pakete abfragen:
+
+`nix-env -qa {{Suche_Begriff}}`
+
+- Paket installieren:
+
+`nix-env -iA nixpkgs.{{pkg_name}}`
```
Comment (EmilySeville7cfg) [inline pages.de/common/nix-env.md:20]: The following fix could be applied after the [original](https://github.com/tldr-pages/tldr/blob/main/pages/common/nix-env.md) page is updated:

```suggestion
`nix-env -iA nixpkgs.{{Paketnamen}}`
```

- Just well known contractions should be used.
- Our style guide doesn't specify whether this fix should be applied or not.

File: pages.de/common/nix-env.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+- Installieren eines Pakets von einer URL:
+
+`nix-env -i {{pkg_name}} --datei {{beispiel.com}}`
```
Comment (EmilySeville7cfg) [inline pages.de/common/nix-env.md:24]: The following fix could be applied after the [original](https://github.com/tldr-pages/tldr/blob/main/pages/common/nix-env.md) page is updated:

```suggestion
`nix-env -i {{Paketnamen}} --datei {{beispiel.com}}`
```

- Just well known contractions should be used.
- Our style guide doesn't specify whether this fix should be applied or not.

File: pages.de/common/nix-env.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+- Paket deinstallieren:
+
+`nix-env -e {{pkg_name}}`
```
Comment (EmilySeville7cfg) [inline pages.de/common/nix-env.md:28]: The following fix could be applied after the [original](https://github.com/tldr-pages/tldr/blob/main/pages/common/nix-env.md) page is updated:

```suggestion
`nix-env -e {{Paketnamen}}`
```

- Just well known contractions should be used.
- Our style guide doesn't specify whether this fix should be applied or not.

File: pages.de/common/nix-env.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+- Upgrade eines Pakets:
+
+`nix-env -u {{pkg_name}}`
```
Comment (EmilySeville7cfg) [inline pages.de/common/nix-env.md:32]: The following fix could be applied after the [original](https://github.com/tldr-pages/tldr/blob/main/pages/common/nix-env.md) page is updated:

```suggestion
`nix-env -u {{Paketnamen}}`
```

- Just well known contractions should be used.
- Our style guide doesn't specify whether this fix should be applied or not.

File: pages.de/common/nix-shell.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+> Startet eine interaktive Shell, die auf einem Nix-Ausdruck basiert.
+> Weitere Informationen: <https://nixos.org/manual/nix/stable/#sec-nix-shell>.
+
+- Start mit Nix-Ausdruck in `shell.nix` oder `default.nix` im aktuellen Verzeichnis:
+
+`nix-shell`
+
+- Shell-Befehl in nicht-interaktiver Shell ausführen und beenden:
+
+`nix-shell --run "{{Befehl}} {{Befehl_Argumente}}"`
```
Comment (EmilySeville7cfg) [inline pages.de/common/nix-shell.md:12]: The following fix could be applied after the [original](https://github.com/tldr-pages/tldr/blob/main/pages/common/nix-shell.md) page is updated:

```suggestion
`nix-shell --run "{{Befehl}} {{arg1 arg2 ...}}"`
```

Wrong syntax is used for zero or more arguments.

File: pages.de/common/nix-shell.md [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+
+- Start mit Ausdruck in `default.nix` im aktuellen Verzeichnis:
+
+`nix-shell {{default.nix}}`
+
+- Start mit aus nixpkgs geladenen Paketen:
+
+`nix-shell --packages {{Paket_name_1}} {{Paket_name_2}}`
```
Comment (EmilySeville7cfg) [inline pages.de/common/nix-shell.md:20]: The following fix could be applied after the [original](https://github.com/tldr-pages/tldr/blob/main/pages/common/nix-shell.md) page is updated:

```suggestion
`nix-shell --packages {{Paket_name_1 Paket_name_2 ...}}`
```

Wrong syntax is used for zero or more arguments.

File: pages.de/common/nix-shell.md [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+
+- Start mit Paketen, die aus einer bestimmten Nixpkgs-Revision geladen wurden:
+
+`nix-shell --packages {{Paket_name}} -I nixpkgs={{https://github.com/NixOS/nixpkgs/archive/nixpkgs_revision.tar.gz}}`
```
Comment (EmilySeville7cfg) [inline pages.de/common/nix-shell.md:24]: The following fix could be applied after the [original](https://github.com/tldr-pages/tldr/blob/main/pages/common/nix-shell.md) page is updated:

```suggestion
`nix-shell --packages {{Paket_name_1 Paket_name_2 ...}} -I nixpkgs={{https://github.com/NixOS/nixpkgs/archive/nixpkgs_revision.tar.gz}}`
```

Wrong syntax is used for zero or more arguments.

File: pages.de/common/nix-shell.md [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+
+- Den Rest der Datei mit einem bestimmten Interpreter auswerten, zur Verwendung in `#!-scripts` (siehe <https://nixos.org/manual/nix/stable/#use-as-a-interpreter>):
+
+`nix-shell -i {{interpreter}} --packages {{package_names}}`
```
Comment (EmilySeville7cfg) [inline pages.de/common/nix-shell.md:28]: The following fix could be applied after the [original](https://github.com/tldr-pages/tldr/blob/main/pages/common/nix-shell.md) page is updated:

```suggestion
`nix-shell -i {{interpreter}} --packages {{Paket_name_1 Paket_name_2 ...}}`
```

Wrong syntax is used for zero or more arguments.

File: pages.de/common/nix.md [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+> Dienstprogramme für die Nix-Sprache und den Nix-Speicher.
+> Weitere Informationen: <https://nixos.org>.
+
+- Suche nach einem Paket über seinen Namen oder seine Beschreibung:
+
+`nix search {{search_term}}`
+
+- Starten einer Nix-Shell, die die angegebenen Pakete zur Verfügung stellt:
+
+`nix run {{nixpkgs.pkg1 nixpkgs.pkg2 nixpkgs.pkg3...}}`
```
Comment (EmilySeville7cfg) [inline pages.de/common/nix.md:12]: The following fix could be applied after the [original](https://github.com/tldr-pages/tldr/blob/main/pages/common/nix.md) page is updated:

```suggestion
`nix run {{nixpkgs.pkg1 nixpkgs.pkg2 ...}}`
```

Wrong syntax is used for zero or more arguments.

**Review by EmilySeville7cfg — CHANGES_REQUESTED** [review]
- Changes were requested because several mistakes were found.
- Several cases are not handled by our style guide but should.
- PR should be reviewed by native speakers to check descriptions for grammar and syntax errors.
- Thanks for a contribution. 😄

File: pages.de/common/nix-shell.md (hunk 11 above)
Comment (kbdharun) [reply pages.de/common/nix-shell.md:28]:
```suggestion
`nix-shell -i {{interpreter}} --packages {{paket_name_1 paket_name_2 ...}}`
```
@EmilySeville7cfg I don't think we should capitalize **P** here. @pixelcmtd @marchersimon Can you check this out?

File: pages.de/common/nix-shell.md (hunk 10 above)
Comment (kbdharun) [reply pages.de/common/nix-shell.md:24]:
```suggestion
`nix-shell --packages {{paket_name_1 paket_name_2 ...}} -I nixpkgs={{https://github.com/NixOS/nixpkgs/archive/nixpkgs_revision.tar.gz}}`
```

File: pages.de/common/nix-shell.md (hunk 9 above)
Comment (kbdharun) [reply pages.de/common/nix-shell.md:20]: Same here
```suggestion
`nix-shell --packages {{paket_name_1 paket_name_2 ...}}`
```

**Comment by rockofox** [comment]
Sorry, I am not quite familiar with the process here. Should I make the changes to the original pages, or is that someone else's responsibility?

**Comment by EmilySeville7cfg** [comment]
Anybody else can modify the original pages. :)

**Comment by EmilySeville7cfg** [comment]
There are several mistakes now:
- `<nixpkgs>` placeholder syntax is unsupported: use `{{package_name1 package_name2 ...}}` (translated) instead.

**Comment by rockofox** [comment]
It's not a placeholder. See https://nixos.org/manual/nix/stable/command-ref/nix-build.html

**Review by EmilySeville7cfg — APPROVED** [review]

File: pages.de/common/nix-collect-garbage.md [hunk 13]
```diff
@@ -0,0 +1,17 @@
+# nix-collect-garbage
+
+> Löschen von unbenutzten und unerreichbaren Nix-Speicherpfaden.
+> Generationen können mit `nix-env --list-generations` aufgelistet werden.
+> Mehr Informationen: <https://nixos.org/releases/nix/latest/manual/#sec-nix-collect-garbage>.
```
Comment (kbdharun) [inline pages.de/common/nix-collect-garbage.md:5]:
```suggestion
> Weitere Informationen: <https://nixos.org/releases/nix/latest/manual/#sec-nix-collect-garbage>.
```

**Review by kbdharun — APPROVED** [review]

**Pull request merged**